### PR TITLE
reorganize APIs (was: seed custom user fields)

### PR DESCRIPTION
**Pull request by berleant**

### CODE REVIEW

File: api/campaign.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  tag_names.forEach(tn => req.field('tag_names[]', tn));
+
+  return req.then(({ body }) => {
+    body;
+    return body;
+  });
+});
+
+const getExistingCampaigns = getExistingTagGroups;
```
Comment (sarayourfriend) [inline api/campaign.js:30]: Cool, I like this aliasing the functions with our own data model on top. Self documenting 🙂

File: api/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const collective = require('./collective');
+const campaign = require('./campaign');
+const topic = require('./topic');
+const user = require('./user');
+
+module.exports = Object.assign(collective, campaign, topic, user);
```
Comment (sarayourfriend) [inline api/index.js]: If you change this to be:

```javascript
module.exports = {
  collective: require('./collective'),
  campaign: require('./campaign'),
  topic: require('./topic'),
  user: require('./user')
};
```

We'll get better type inference from VS code for free.

File: api/user.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+const getCustomUserFields = (exports.getCustomUserFields = () =>
+  sa
+    .get('http://localhost:3000/admin/customize/user_fields.json')
+    .query({ api_key, api_username })
+    .then(({ body: { user_fields } }) => user_fields));
+
+const createCustomUserField = (exports.createCustomUserField = async user_field => {
+  console.assert(['text', 'confirm', 'dropdown'].includes(user_field.field_type));
+
+  let options_string;
```
Comment (sarayourfriend) [inline api/user.js]: Unless we're passing the variable into the fields thing, let's keep with the idiomatic camel casing in JavaScript. The only reason I had the API key and username like that was to make it convenient to put into the field object, but adding those to a request should probably be pushed off into a utility somewhere so we don't have to write the same query string and body code everywhere.

File: seed.js [hunk 4]
```diff
@@ -1,116 +1,141 @@
...
 We are committed to direct action, mutual aid and campaign support.`,
-    },
   },
 };
+
+const customUserFields = [
+  {
+    name: 'Debt Amount',
+    description: 'A number greater than 0',
+    field_type: 'text',
```
Comment (sarayourfriend) [inline seed.js]: Can you add a comment that this needs to be text for us to support the decimal? I'm guessing that's why we can't use the `:integer` type

File: api/campaign.js (hunk 1 above)
Comment (berleant) [reply api/campaign.js:30]: Whoops, actually I meant to remove those lines/rename the actual functions. But if you like it, let's keep it.

File: api/user.js (hunk 3 above)
Comment (berleant) [reply api/user.js]: Ah, that's why I was confused. Sounds good.

File: api/index.js (hunk 2 above)
Comment (berleant) [reply api/index.js]: cool

File: seed.js (hunk 4 above)
Comment (berleant) [reply seed.js]: actually because discourse doesn't have that option, only 'dropdown', 'text', and 'confirm' (whatever that is)

**Comment by berleant** [comment]
I decided to use the plugin to make the custom fields so this PR is basically refactoring now + a few extra APIs

File: api/user.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      return users[0];
+    }));
+
+const getCustomUserFields = (exports.getCustomUserFields = () =>
+  sa
+    .get('http://localhost:3000/admin/customize/user_fields.json')
+    .query({ api_key, api_username })
+    .then(({ body: { user_fields } }) => user_fields));
+
+const createCustomUserField = (exports.createCustomUserField = async user_field => {
```
Comment (sarayourfriend) [inline api/user.js:23]: So we're not actually using this method anymore, right?

**Comment by sarayourfriend** [comment]
LGTM

**Review by sarayourfriend — APPROVED** [review]

**Pull request merged**